partiql-plan/src/main/resources/partiql_plan.ion

```diff
@@ -300,6 +300,8 @@ rel::{
       projections:  list::[rex],
     },
 
+    // TODO: Specify that this is a LATERAL JOIN. Create a separate JOIN. Also, determine the allowable types of JOIN.
+    // For context: Oracle SQL doesn't allow ... FULL OUTER JOIN LATERAL ... or ... RIGHT OUTER JOIN LATERAL ...
```

Comment:
PartiQL Spec section 5.7

SQL 2003 used the LATERAL keyword to correlate FROM clause items. In the interest of compatibility with SQL, PartiQL also allows the use of the keyword LATERAL, though it does not do anything more than the comma itself would do.
That is “l , LATERAL r” is equivalent to “l , r”.


This is: I don't think we need to support LATERAL JOIN. The spec is quite clear about there is no difference between left join and left lateral join, inner join and inner lateral join. 

The spec is also quite clear about the fact that Full Join can not be lateral. 

I think the only decision we have to make here is if we want right join to be lateral. 

I vote no : )

Reply:
Discussed offline a while back. I'd like to have different plan nodes for both types -- which corresponds to a different implementation in the evaluator. While yes, syntactically, one doesn't need to use the LATERAL keyword, it has semantic implementations. Upon analyzing the plan and seeing no references to bindings from the LHS, we can potentially use the normal join operator (and not the lateral join).

If we ever want to support that in the future, what would we call this plan node in the future? `non_lateral_join`? Since `join` is already taken and is used to semantically denote a lateral join?

Especially given that we are a library where DB implementers choose which features to support. Does their execution allow for joins? DynamoDB doesn't. What about lateral joins? Does every implementation support lateral joins in their execution? I'm not sure -- but I do know that lateral joins are expensive due to their correlated nature. I believe it limits the implementations that one can use. If I'm not mistaken, you are forced to use a variation of a nested loop join.